deployment/HDI-Spark/Notebooks/README.md

```diff
@@ -0,0 +1,116 @@
+Deploy Spark .Net REPL to HDInsight Spark cluster with Notebook Experience
+===
+
+# Table of Contents
+This documentation will elaborate the steps on how to deploy Spark .Net REPL to HDI Spark Cluster and submit jobs through Jupyter Notebook using Spark .Net.  
+- [Background](#background)
+- [Pre-requisites](#pre-requisites)
+- [Steps to deploy Spark .Net REPL to HDInsight Spark cluster](#steps-to-deploy-spark-net-repl-to-hdinsight-spark-cluster)
+  - [Step 1. Stop Livy Server](#step-1-stop-livy-server)
+  - [Step 2. Submit HDInsight Script Action](#step-2-submit-hdinsight-script-action)
+  - [Step 3. Start Livy Server](#step-3-start-livy-server)
+   - [Step 4. Set up Spark Default Configs](#step-3-start-livy-server)
+- [Submit Jobs through Jupyter Notebook](#submit-jobs-through-jupyter-notebook)
+
+## Background
+We are currently using [dotnet-try](https://github.com/dotnet/try) as our dotnet REPL. ` dotnet-try ` can be used as a jupyter kernel and we implement it through [livy](https://github.com/apache/incubator-livy) and [sparkmagic](https://github.com/jupyter-incubator/sparkmagic). To enable Jupyter Notebook Experience using Spark .Net, this will include simple manual steps through [Ambari](https://docs.microsoft.com/en-us/azure/hdinsight/hdinsight-hadoop-manage-ambari) and submit [script actions](https://docs.microsoft.com/en-us/azure/hdinsight/hdinsight-hadoop-customize-cluster-linux) on HDInsight Spark cluster.
```

Comment:
```suggestion
To provide the ability to use Spark .NET within your Jupyter Notebooks, an important pre-requisite is a C# REPL that is capable of executing your C# code line-by-line (and preserving execution state, where needed). We have integrated [dotnet-try](https://github.com/dotnet/try) as our official dotnet REPL to provide you with the best experience. 

To provide the end-to-end integration, we had to add a Jupyter kernel along with necessary modifications to [Apache Livy](https://github.com/apache/incubator-livy) and [sparkmagic](https://github.com/jupyter-incubator/sparkmagic). Since these code changes have not yet been merged into the respective open source projects, to be able to get the out-of-box Spark .NET experience, you will have to patch components in your HDInsight Spark Cluster with the binaries we provide. Please note that this is *experimental* and not supported by the HDInsight Spark team. *Use it at your own risk*. Please do that we are working hard to get these changes into the respective projects so you do not need these manual steps.

To enable Spark .NET through the Jupyter Notebook Experience, you would have to follow a few manual steps through [Ambari](https://docs.microsoft.com/en-us/azure/hdinsight/hdinsight-hadoop-manage-ambari) and submit [script actions](https://docs.microsoft.com/en-us/azure/hdinsight/hdinsight-hadoop-customize-cluster-linux) on HDInsight Spark cluster.
```